src/cr/cube/dimension.py

```diff
@@ -922,9 +927,17 @@ def translate_element_id(self, _id) -> Optional[str]:
            0-# of elements), use _id'th alias.
         6) If all of these fail, return None.
         """
-        if self._dimension_type not in DT.ARRAY_TYPES:
+        if self._dimension_type not in DT.SHIMMED_TYPES:
             return _id
 
+        if self._dimension_type == DT.DATETIME:
+            try:
+                int_id = int(_id)
+            except ValueError:
+                int_id = _id
```

Comment:
is this try block used to check is `_id` is a str representation for a numeric value? If yes please use `str.isnumeric()` instead of using that try block